src/network/core.rs

```diff
@@ -520,6 +528,12 @@ impl Core {
                 for ip in &netns_ipaddr {
                     netns_ipaddr_clone.push(*ip)
                 }
+
+                let mut macvlan_gateway = gw_ipaddr;
+                if internal {
+                    // force empty gateway if macvlan is marked as internal
+                    macvlan_gateway = Vec::new();
+                }
```

Comment:
This should just be added to line 406 - 435, no reason to overwrite here.

Reply:
yes lets do it there.